src/list/list.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Controller from '../controller.js';
+import ControllerCollection from '/ckeditor5/ui/controllercollection.js';
+
+import ListItemView from './listitemview.js';
+
+/**
+ * The basic list controller class.
+ *
+ * @memberOf ui.list
+ * @extends ui.Controller
+ */
+
+export default class List extends Controller {
+	/**
+	 * Creates a List instance.
+	 *
+	 * @param {utils.Observable} model
+	 * @param {ui.View} view
+	 */
+	constructor( model, view ) {
+		super( model, view );
+
+		this.collections.add( new ControllerCollection( 'list' ) );
+	}
+
+	init() {
+		// Initially populate "list" controller collection with children from model.items.
+		for ( let itemModel of this.model.items ) {
```

Comment:
Let's report a followup for this. It doesn't make sense to work on simplifying this code just yet, but once e.g. toolbar model will be improved, we need to reconsider how to do this.